### PR TITLE
opae-libs: add DFL driver interrupt support

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: plugins/xfpga/event.c [hunk 1]
```diff
@@ -166,43 +172,38 @@ STATIC fpga_result send_uafu_event_request(fpga_handle handle,
 {
 	int res = FPGA_OK;
 	int fd = FILE_DESCRIPTOR(event_handle);
-	struct _fpga_event_handle *_eh =
-		(struct _fpga_event_handle *)event_handle;
 	struct _fpga_handle *_handle = (struct _fpga_handle *)handle;
-	opae_port_info port_info = { 0 };
+	uint32_t num_irqs;
```
Comment (tswhison) [inline plugins/xfpga/event.c:176]: Initialize num_irqs to 0.

File: plugins/xfpga/event.c [hunk 2]
```diff
@@ -220,8 +221,7 @@ STATIC fpga_result check_interrupts_supported(fpga_handle handle,
 	fpga_result destroy_res = FPGA_OK;
 	fpga_properties prop = NULL;
 	struct _fpga_handle *_handle = (struct _fpga_handle *)handle;
-	opae_fme_info fme_info = { 0 };
-	opae_port_info port_info = { 0 };
+	uint32_t num_irqs;
```
Comment (tswhison) [inline plugins/xfpga/event.c:224]: Initialize num_irqs to 0.

File: plugins/xfpga/opae_drv.c [hunk 3]
```diff
@@ -434,6 +448,86 @@ fpga_result dfl_fme_port_reset(int fd)
 	return opae_ioctl(fd, DFL_FPGA_PORT_RESET, NULL);
 }
 
+fpga_result dfl_port_get_err_irq(int fd, uint32_t *num_irqs)
+{
+	ASSERT_NOT_NULL(num_irqs);
+	int res = opae_ioctl(fd, DFL_FPGA_PORT_ERR_GET_IRQ_NUM, num_irqs);
+	if (!res) {
+		OPAE_MSG("Ioctl DFL_FPGA_PORT_ERR_GET_IRQ_NUM error=%d", res);
```
Comment (tswhison) [inline plugins/xfpga/opae_drv.c:456]: Shouldn't these be OPAE_ERR?

File: plugins/xfpga/opae_drv.c [hunk 4]
```diff
@@ -434,6 +448,86 @@ fpga_result dfl_fme_port_reset(int fd)
...
+	}
+	return res;
+}
+
+fpga_result dfl_port_get_user_irq(int fd, uint32_t *num_irqs)
+{
+	ASSERT_NOT_NULL(num_irqs);
+	int res = opae_ioctl(fd, DFL_FPGA_PORT_UINT_GET_IRQ_NUM, num_irqs);
+	if (!res) {
+		OPAE_MSG("Ioctl DFL_FPGA_PORT_UINT_GET_IRQ_NUM error=%d", res);
```
Comment (tswhison) [inline plugins/xfpga/opae_drv.c:466]: OPAE_ERR?

File: plugins/xfpga/opae_drv.c [hunk 5]
```diff
@@ -434,6 +448,86 @@ fpga_result dfl_fme_port_reset(int fd)
...
+	}
+	return res;
+}
+
+fpga_result dfl_fme_get_err_irq(int fd, uint32_t *num_irqs)
+{
+	ASSERT_NOT_NULL(num_irqs);
+	int res = opae_ioctl(fd, DFL_FPGA_FME_ERR_GET_IRQ_NUM, num_irqs);
+	if (!res) {
+		OPAE_MSG("Ioctl DFL_FPGA_FME_ERR_GET_IRQ_NUM error=%d", res);
```
Comment (tswhison) [inline plugins/xfpga/opae_drv.c:476]: OPAE_ERR?

File: plugins/xfpga/opae_drv.c [hunk 6]
```diff
@@ -434,6 +448,86 @@ fpga_result dfl_fme_port_reset(int fd)
...
+	}
+	return res;
+}
+
+fpga_result dfl_set_irq(int fd,uint32_t start,
+	uint32_t count, int32_t *eventfd,int ioctl_id)
```
Comment (tswhison) [inline plugins/xfpga/opae_drv.c:482]: Spacing is off.

Comment (anandaravuri) [reply plugins/xfpga/opae_drv.c]: fixed

File: plugins/xfpga/opae_drv.c (hunk 5 above)
Comment (anandaravuri) [reply plugins/xfpga/opae_drv.c]: fixed

File: plugins/xfpga/opae_drv.c (hunk 4 above)
Comment (anandaravuri) [reply plugins/xfpga/opae_drv.c]: fixed

File: plugins/xfpga/opae_drv.c (hunk 3 above)
Comment (anandaravuri) [reply plugins/xfpga/opae_drv.c]: fixed

File: plugins/xfpga/event.c (hunk 2 above)
Comment (anandaravuri) [reply plugins/xfpga/event.c]: fixed

File: plugins/xfpga/event.c (hunk 1 above)
Comment (anandaravuri) [reply plugins/xfpga/event.c]: fixed

**Pull request merged**